### ISSUE TITLE
Handle Signup 

### ISSUE THREAD

**Issue by nrhubbar** [common-threads-group/api-server#7]
* [x] Create new route for `POST /signup`
* [ ] Add new user to User Collection in MongoDB
* [ ] Add and Associate new Profile entry in MongoDB

**Comment by nrhubbar** [comment]
1 once @herrejes code is merged into `Dev`, do a `git pull`
2 `git checkout Dev`
3 `git checkout -b ArinMauk-issue7-handleSignup`
4 do work, commit often, push less often
5 start pull request into `Dev`
